data/actions/scripts/other/potions.lua

```diff
@@ -1,158 +1,219 @@
+						return false
+					end
+
+					for k, v in ipairs(potion.vocations) do
+						if not tablefind(potion.vocations, Vocation(v):getDemotion():getId()) then
+							table.insert(vocations, v)
+						end
+					end
+				end
+
+				for i = 1, #vocations do
+					if i == #vocations then
+						str = str .. (#vocations ~= 1 and " and " or "") .. Vocation(vocations[i]):getName():lower() .. "s"
+					else
+						str = str .. (i ~= 1 and ", " or "") .. Vocation(vocations[i]):getName():lower() .. "s"
+					end
+				end
+
+				vocations = nil
+
+				if str == "" then
+					str = "players"
+				end
+
+				player:say(string.format(potions.stringOfVocation, str, potion.minLevel), TALKTYPE_MONSTER_SAY)
+				return false
+			end
+		end
+
+		-- Condition exists after the check!
+		if player:getCondition(CONDITION_EXHAUST_HEAL) then
+			player:sendTextMessage(MESSAGE_STATUS_SMALL, Game.getReturnMessage(RETURNVALUE_YOUAREEXHAUSTED))
 			return true
 		end
 
-		if not doTargetCombatHealth(0, target, COMBAT_HEALING, 250, 350, CONST_ME_MAGIC_BLUE) then
-			return false
+		if potion.healthToAdd then
+			if not doTargetCombatHealth((target.uid == player.uid and player.uid or 0), target, COMBAT_HEALING, potion.healthToAdd.min * (potions.healthRate and potions.healthRate or 1),
+					potion.healthToAdd.max * (potions.healthRate and potions.healthRate or 1), CONST_ME_MAGIC_BLUE) then
+				return false
+			end
 		end
 
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(strongEmptyPot, 1)
-	elseif itemId == strongManaPot then
-		if (not isInArray({1, 2, 3, 5, 6, 7}, target:getVocation():getId()) or target:getLevel() < 50) and not getPlayerFlagValue(player, PlayerFlag_IgnoreSpellCheck) then
-			player:say("This potion can only be consumed by sorcerers, druids and paladins of level 50 or higher.", TALKTYPE_MONSTER_SAY)
-			return true
+		if potion.manaToAdd then
+			if not doTargetCombatMana((target.uid == player.uid and player.uid or 0), target, potion.manaToAdd.min * (potions.manaRate and potions.manaRate or 1),
+					potion.manaToAdd.max * (potions.manaRate and potions.manaRate or 1), CONST_ME_MAGIC_BLUE) then
+				return false
+			end
 		end
 
-		if not doTargetCombatMana(0, target, 115, 185, CONST_ME_MAGIC_BLUE) then
-			return false
+		if potion.combat then
+			if not potion.combat:execute(target, numberToVariant(target:getId())) then
+				return false
+			end
 		end
 
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(strongEmptyPot, 1)
-	elseif itemId == greatSpiritPot then
-		if (not isInArray({3, 7}, target:getVocation():getId()) or target:getLevel() < 80) and not getPlayerFlagValue(player, PlayerFlag_IgnoreSpellCheck) then
-			player:say("This potion can only be consumed by paladins of level 80 or higher.", TALKTYPE_MONSTER_SAY)
-			return true
+		if potion.condition and potiom.condition:getType() ~= CONDITION_EXHAUST_HEAL then
```

Comment:
The issue with getTopParent() happened because i forgot to change "potion" to "item"
I tested it right now, everything worked!
I don't intend more updates since i can't test all the time ( exams ).